---
title: "GWAS data analysis of family history of Alzheimer's disease"
author: "Haoxue Sun"
output: 
  html_document:
    toc: false 
    depth: 3 
    theme: paper 
    highlight: tango
---
```{r set-options, echo=FALSE, cache=FALSE}
options(width = 400)
```  
***

### Overview

There are studies using Genome-wide meta-analysis (GWAS) to analyse family history of Alzheimer's disease and identify new loci and functional pathways influencing Alzheimer's disease risk. However, the sample number of each study is under 500,000. By putting together data from 6 different studies, I plan to look into whether common conclusions could be drawn from them, and if there is difference when comparing maternal and paternal history. I use data from 6 different studies posted on [GWAS Catalog](https://www.ebi.ac.uk/gwas/efotraits/EFO_0009268), with a sample number of 9,653,950 SNPs in total. The study identified several genes, including NECTIN2 and AC011481.1, as AD-associated loci that have known and putative biological processes associated with Alzheimerâ€™s disease. Using published consortium data and clinically diagnosed AD and AD-by-proxy data, 32 risk loci were identified in total, implicating 247 potential causative genes.

Three faculties:
Blanca Himes, PhD, Assistant Professor, Informatics in Biostatistics and Epidemiology
Shen Li, PhD, Professor, Informatics in Biostatistics and Epidemiology
Xiaohui Yao, PhD, PostDoc, Informatics in Biostatistics and Epidemiology

[Link to final project Github repository](https://github.com/shx0426/BMIN503_Final_Project)


### Introduction 

Alzheimer's disease (AD) has become one of the public health priorities for the 21st century. Recent studies have shown that AD is highly heritable, and over 20 AD-associated locis have been identified. There are studies using Genome-wide meta-analysis (GWAS) to analyse family history of Alzheimer's disease and identify new loci and functional pathways influencing Alzheimer's disease risk. There are summary statistics result of GWAS with 314,278 participants from UK Biobank (27,696 maternal cases, 14,338 paternal cases), and also GWAS of clinically diagnosed AD and AD-by-proxy (71,880 cases, 383,378 controls) in Europe. Aggregating the results from different studies is a step forward in identifying the genetic factors that contribute to AD risk based on a larger scale. Also, the possible difference between maternal and paternal AD history.

AD is now the most frequent neurodegenerative disease, since about 35 million people have been diagosed. There are millions of clinical studies along with lab researches going on foucsing on AD. The data yield from clinical records and lab studies then requires effective analysis using statistics and informatics skills. Hypothesis can be proposed from the result of data analysis, and further experiment can be carried out in labs. Admittedly, the raw data in the open database could contain unneglectable noise, resulting in the potential biased outcome. Large sample size and sufficient experiment tests can help to develop the reliability of the result.


### Methods

Data from UK Biobank cohort and GWAS catalog were utilized to enlarge the total sample sacle. GWAS, also known as whole genome association study (WGA study, or WGAS), is an observational study of a genome-wide set of genetic variants in different individuals to see if any variant is associated with a trait. GWASs typically focus on associations between single-nucleotide polymorphisms (SNPs) and traits like major human diseases, but can equally be applied to any other genetic variants and any other organisms.


```{r}

AD_sumstats_1 <- read.table(file = "1_UK_Biobank_maternal_17May2018.txt", #Name of file
                header = TRUE, 
                sep = " ", #Character that separates  column entries (tab, space, comma)
                skip = 0, #Number of lines to skip from beginning
                as.is = TRUE, #Should character variables be left alone 
                na.strings = NA, #The encoding of missing values. Blanks are assumed missing too.
                comment.char = "#", #Lines of file beginning with this character are ignored. (# default)
                quote = "", #Character used to quote values. "" is no quotes.
                nrows = -1 #The number of rows to read in. Negative/invalid values are ignored
)

AD_sumstats_2 <- read.table(file = "2_UK_Biobank_paternal_17May2018.txt", 
                header = TRUE, 
                sep = " ", 
                skip = 0, 
                as.is = TRUE,  
                na.strings = NA, 
                comment.char = "#",
                quote = "", 
                nrows = -1 
)

AD_sumstats_3 <- read.table(file = "3_UK_Biobank_parental_17May2018.txt", 
                header = TRUE, 
                sep = " ", 
                skip = 0, 
                as.is = TRUE, 
                na.strings = NA, 
                comment.char = "#", 
                quote = "", 
                nrows = -1 
)

AD_sumstats_4 <- read.table(file = "4_UK_Biobank_IGAP_17May2018.txt", 
                header = TRUE, 
                sep = " ", 
                skip = 0, 
                as.is = TRUE, 
                na.strings = NA, 
                comment.char = "#", 
                quote = "", 
                nrows = -1 
)

AD_sumstats <- read.table(file = "GWAS data.txt", 
                header = TRUE, 
                sep = "", 
                skip = 0, 
                as.is = TRUE, 
                na.strings = NA, 
                comment.char = "#", 
                quote = "", 
                nrows = -1 
)

library(ggplot2)
library(dplyr)
library(lattice)
manhattan.plot<-function(chr, pos, pvalue, 
	sig.level=NA, annotate=NULL, ann.default=list(),
	should.thin=T, thin.pos.places=2, thin.logp.places=2, 
	xlab="Chromosome", ylab=expression(-log[10](p-value)),
	col=c("gray","darkgray"), panel.extra=NULL, pch=20, cex=0.8,...) {

	if (length(chr)==0) stop("chromosome vector is empty")
	if (length(pos)==0) stop("position vector is empty")
	if (length(pvalue)==0) stop("pvalue vector is empty")

	#make sure we have an ordered factor
	if(!is.ordered(chr)) {
		chr <- ordered(chr)
	} else {
		chr <- chr[,drop=T]
	}

	#make sure positions are in kbp
	if (any(pos>1e6)) pos<-pos/1e6;

	#calculate absolute genomic position
	#from relative chromosomal positions
	posmin <- tapply(pos,chr, min);
	posmax <- tapply(pos,chr, max);
	posshift <- head(c(0,cumsum(posmax)),-1);
	names(posshift) <- levels(chr)
	genpos <- pos + posshift[chr];
	getGenPos<-function(cchr, cpos) {
		p<-posshift[as.character(cchr)]+cpos
		return(p)
	}

	#parse annotations
	grp <- NULL
	ann.settings <- list()
	label.default<-list(x="peak",y="peak",adj=NULL, pos=3, offset=0.5, 
		col=NULL, fontface=NULL, fontsize=NULL, show=F)
	parse.label<-function(rawval, groupname) {
		r<-list(text=groupname)
		if(is.logical(rawval)) {
			if(!rawval) {r$show <- F}
		} else if (is.character(rawval) || is.expression(rawval)) {
			if(nchar(rawval)>=1) {
				r$text <- rawval
			}
		} else if (is.list(rawval)) {
			r <- modifyList(r, rawval)
		}
		return(r)
	}

	if(!is.null(annotate)) {
		if (is.list(annotate)) {
			grp <- annotate[[1]]
		} else {
			grp <- annotate
		} 
		if (!is.factor(grp)) {
			grp <- factor(grp)
		}
	} else {
		grp <- factor(rep(1, times=length(pvalue)))
	}
  
	ann.settings<-vector("list", length(levels(grp)))
	ann.settings[[1]]<-list(pch=pch, col=col, cex=cex, fill=col, label=label.default)

	if (length(ann.settings)>1) { 
		lcols<-trellis.par.get("superpose.symbol")$col 
		lfills<-trellis.par.get("superpose.symbol")$fill
		for(i in 2:length(levels(grp))) {
			ann.settings[[i]]<-list(pch=pch, 
				col=lcols[(i-2) %% length(lcols) +1 ], 
				fill=lfills[(i-2) %% length(lfills) +1 ], 
				cex=cex, label=label.default);
			ann.settings[[i]]$label$show <- T
		}
		names(ann.settings)<-levels(grp)
	}
	for(i in 1:length(ann.settings)) {
		if (i>1) {ann.settings[[i]] <- modifyList(ann.settings[[i]], ann.default)}
		ann.settings[[i]]$label <- modifyList(ann.settings[[i]]$label, 
			parse.label(ann.settings[[i]]$label, levels(grp)[i]))
	}
	if(is.list(annotate) && length(annotate)>1) {
		user.cols <- 2:length(annotate)
		ann.cols <- c()
		if(!is.null(names(annotate[-1])) && all(names(annotate[-1])!="")) {
			ann.cols<-match(names(annotate)[-1], names(ann.settings))
		} else {
			ann.cols<-user.cols-1
		}
		for(i in seq_along(user.cols)) {
			if(!is.null(annotate[[user.cols[i]]]$label)) {
				annotate[[user.cols[i]]]$label<-parse.label(annotate[[user.cols[i]]]$label, 
					levels(grp)[ann.cols[i]])
			}
			ann.settings[[ann.cols[i]]]<-modifyList(ann.settings[[ann.cols[i]]], 
				annotate[[user.cols[i]]])
		}
	}
 	rm(annotate)

	#reduce number of points plotted
	if(should.thin) {
		thinned <- unique(data.frame(
			logp=round(-log10(pvalue),thin.logp.places), 
			pos=round(genpos,thin.pos.places), 
			chr=chr,
			grp=grp)
		)
		logp <- thinned$logp
		genpos <- thinned$pos
		chr <- thinned$chr
		grp <- thinned$grp
		rm(thinned)
	} else {
		logp <- -log10(pvalue)
	}
	rm(pos, pvalue)
	gc()

	#custom axis to print chromosome names
	axis.chr <- function(side,...) {
		if(side=="bottom") {
			panel.axis(side=side, outside=T,
				at=((posmax+posmin)/2+posshift),
				labels=levels(chr), 
				ticks=F, rot=0,
				check.overlap=F
			)
		} else if (side=="top" || side=="right") {
			panel.axis(side=side, draw.labels=F, ticks=F);
		}
		else {
			axis.default(side=side,...);
		}
	 }

	#make sure the y-lim covers the range (plus a bit more to look nice)
	prepanel.chr<-function(x,y,...) { 
		A<-list();
		maxy<-ceiling(max(y, ifelse(!is.na(sig.level), -log10(sig.level), 0)))+.5;
		A$ylim=c(0,maxy);
		A;
	}

	xyplot(logp~genpos, chr=chr, groups=grp,
		axis=axis.chr, ann.settings=ann.settings, 
		prepanel=prepanel.chr, scales=list(axs="i"),
		panel=function(x, y, ..., getgenpos) {
			if(!is.na(sig.level)) {
				#add significance line (if requested)
				panel.abline(h=-log10(sig.level), lty=2);
			}
			panel.superpose(x, y, ..., getgenpos=getgenpos);
			if(!is.null(panel.extra)) {
				panel.extra(x,y, getgenpos, ...)
			}
		},
		panel.groups = function(x,y,..., subscripts, group.number) {
			A<-list(...)
			#allow for different annotation settings
			gs <- ann.settings[[group.number]]
			A$col.symbol <- gs$col[(as.numeric(chr[subscripts])-1) %% length(gs$col) + 1]    
			A$cex <- gs$cex[(as.numeric(chr[subscripts])-1) %% length(gs$cex) + 1]
			A$pch <- gs$pch[(as.numeric(chr[subscripts])-1) %% length(gs$pch) + 1]
			A$fill <- gs$fill[(as.numeric(chr[subscripts])-1) %% length(gs$fill) + 1]
			A$x <- x
			A$y <- y
			do.call("panel.xyplot", A)
			#draw labels (if requested)
			if(gs$label$show) {
				gt<-gs$label
				names(gt)[which(names(gt)=="text")]<-"labels"
				gt$show<-NULL
				if(is.character(gt$x) | is.character(gt$y)) {
					peak = which.max(y)
					center = mean(range(x))
					if (is.character(gt$x)) {
						if(gt$x=="peak") {gt$x<-x[peak]}
						if(gt$x=="center") {gt$x<-center}
					}
					if (is.character(gt$y)) {
						if(gt$y=="peak") {gt$y<-y[peak]}
					}
				}
				if(is.list(gt$x)) {
					gt$x<-A$getgenpos(gt$x[[1]],gt$x[[2]])
				}
				do.call("panel.text", gt)
			}
		},
		xlab=xlab, ylab=ylab, 
		panel.extra=panel.extra, getgenpos=getGenPos, ...
	);
}

#select the samples that has p value less than 10^-6
select_sumstats <- filter(AD_sumstats_1, AD_sumstats_1$P < 0.000001)
select_sumstats <- rbind(select_sumstats, filter(AD_sumstats_2, AD_sumstats_2$P < 0.000001))

#visualization of how many SNPs has p value less than 10^-6
ggplot(select_sumstats)+
  geom_point(aes(x = SNP, y = P), color = "blue")+
  scale_y_reverse()

AD_sumstats$Allele <- gsub("?", "", gsub("\\.", "", AD_sumstats$Allele))
AD_sumstats$P.value <- as.numeric(gsub("x10", "e", gsub("\\.", "", AD_sumstats$P.value)))
AD_sumstats$Location <- as.numeric(gsub(":", ".", gsub("\\.", "", AD_sumstats$Location)))
AD_sumstats <- AD_sumstats %>%
  select(Allele, P.value, Mapped.gene, Location)

```

### Results

The following 5 genes have been most frequently apppearing with a small p value (at least < 10^-5). Also a bar plot contains all the genes that has shown p value less than 10^-5 was drawn, among these genes the frequent apearing ones could be potential causative genes for AD, which worth further study.

```{r}
rev(sort(table(AD_sumstats$Mapped.gene)))[1:5]
ggplot(AD_sumstats, aes(x = Mapped.gene))+
  geom_bar()
```

Looking at the two most frequent mapped genes in the studies, NECTIN2 and AC011481.1. NECTIN2 appeared 36 times with p value less than 1*10^-5, it even has 6 alleles that showing p value < 10^-100, which indicates NECTIN2 has high potential of relativeness to AD. AC011481.1 appeared 15 times, containing 3 alleles with p value < 10^-100. Both of the genes and those appeared more than 10 times, demonstrated the possible sites related to AD and could be future study focus.

```{r}
NECTIN2 <- AD_sumstats[which(AD_sumstats$Mapped.gene == "NECTIN2"), ]
NECTIN2

AC011481.1 <- AD_sumstats[which(AD_sumstats$Mapped.gene == "AC011481.1"), ]
AC011481.1
```

The locations of where the frequent mapped genes are is another interesting point. The locations, which indicates which chromosome the gene is located, scattered in half of all the chromosomes. Chromosome 19, contains approximately 60% of the mapped genes, being the chromosome that has the largest nuber of genes related to AD. Other chromosomes also contains a number of genes that is not neglectable. Whether the scattered gene distribution is resulted by AD or mutiple gene function leads to AD remains unknown, but the frequent appeared genes are worth for further study of the relativeness with AD.

```{r}
ggplot(AD_sumstats)+
  geom_point(aes(x = Location, y = Mapped.gene), color = "red")
ggplot(AD_sumstats)+
  geom_point(aes(x = Location, y = P.value), color = "red")+
  scale_y_reverse()
```

### Conclusion

The study identified several genes, including NECTIN2 and AC011481.1, as AD-associated loci that have known and putative biological processes associated with Alzheimerâ€™s disease. In addition, loci ADAM10, BCKDK/KAT8 and ACE contain genes relevant for AD and neurodegeneration. Using published consortium data and clinically diagnosed AD and AD-by-proxy data, 32 risk loci were identified in total, implicating 247 potential causative genes. By bringing together the results from different studies, the database was significantly enlarged. This helps us to further identify the genetic factors that contribute to AD risk based on a larger scale. The potential loci could be drug targets in the future exploration.

