---
title: "BMIN503/EPID600 Project - GWAS data analysis"
author: "Haoxue Sun"
output: 
  html_document:
    toc: false 
    depth: 3 
    theme: paper 
    highlight: tango
---
```{r set-options, echo=FALSE, cache=FALSE}
options(width = 400)
```  
***
Use this template to complete your project throughout the course. Your Final Project presentation in class will be based on the contents of this document. Replace the title/name and text below with your own, but keep the headers.

### Overview
In this section, give a brief a description of your project and its goal, what data you are using to complete it, and what three faculty/staff in different fields you have spoken to about your project with a brief summary of what you learned from each person. Include a link to your final project GitHub repository.

There are studies using Genome-wide meta-analysis (GWAS) to analyse family history of Alzheimer's disease and identify new loci and functional pathways influencing Alzheimer's disease risk. However, the sample number of each study is under 500,000. By putting together data from 6 different studies, I plan to look into whether common conclusions could be drawn from them, and if there is difference when comparing maternal and paternal history.

I will use data from 6 different studies posted on [GWAS Catalog](https://www.ebi.ac.uk/gwas/efotraits/EFO_0009268), with a sample number of 2,0008,882 in total.

Three faculties:
Blanca Himes, PhD, Assistant Professor, Informatics in Biostatistics and Epidemiology
Shen Li, PhD, Professor, Informatics in Biostatistics and Epidemiology
Xiaohui Yao, PhD, PostDoc, Informatics in Biostatistics and Epidemiology

[Link to final project Github repository](https://github.com/shx0426/BMIN503_Final_Project)


### Introduction 
In the first paragraph, describe the problem addressed, its significance, and some background to motivate the problem.
In the second paragraph, explain why your problem is interdisciplinary, what fields can contribute to its understanding, and incorporate background related to what you learned from meeting with faculty/staff.

Alzheimer's disease (AD) has become one of the public health priorities for the 21st century. Recent studies have shown that AD is highly heritable, and over 20 AD-associated locis have been identified. There are studies using Genome-wide meta-analysis (GWAS) to analyse family history of Alzheimer's disease and identify new loci and functional pathways influencing Alzheimer's disease risk. There are summary statistics result of GWAS with 314,278 participants from UK Biobank (27,696 maternal cases, 14,338 paternal cases), and also GWAS of clinically diagnosed AD and AD-by-proxy (71,880 cases, 383,378 controls) in Europe. Aggregating the results from different studies is a step forward in identifying the genetic factors that contribute to AD risk based on a larger scale. Also, the possible difference between maternal and paternal AD history.

AD is now the most frequent neurodegenerative disease, since about 35 million people have been diagosed. There are millions of clinical studies along with lab researches going on foucsing on AD. The data yield from clinical records and lab studies then requires effective analysis using statistics and informatics skills. Hypothesis can be proposed from the result of data analysis, and further experiment can be carried out in labs. Admittedly, the raw data in the open database could contain unneglectable noise, resulting in the potential biased outcome. Large sample size and sufficient experiment tests can help to develop the reliability of the result.


### Methods
In the first paragraph, describe the data used and general methodological approach. Subsequently, incorporate full R code necessary to retrieve and clean data, and perform analysis. Be sure to include a description of code so that others (including your future self) can understand what you are doing and why. 

```{r}

AD_sumstats_1 <- read.table(file = "1_UK_Biobank_maternal_17May2018.txt", #Name of file
                header = TRUE, 
                sep = " ", #Character that separates  column entries (tab, space, comma)
                skip = 0, #Number of lines to skip from beginning
                as.is = TRUE, #Should character variables be left alone 
                na.strings = NA, #The encoding of missing values. Blanks are assumed missing too.
                comment.char = "#", #Lines of file beginning with this character are ignored. (# default)
                quote = "", #Character used to quote values. "" is no quotes.
                nrows = -1 #The number of rows to read in. Negative/invalid values are ignored
)

AD_sumstats_2 <- read.table(file = "2_UK_Biobank_paternal_17May2018.txt", 
                header = TRUE, 
                sep = " ", 
                skip = 0, 
                as.is = TRUE,  
                na.strings = NA, 
                comment.char = "#",
                quote = "", 
                nrows = -1 
)

AD_sumstats_3 <- read.table(file = "3_UK_Biobank_parental_17May2018.txt", 
                header = TRUE, 
                sep = " ", 
                skip = 0, 
                as.is = TRUE, 
                na.strings = NA, 
                comment.char = "#", 
                quote = "", 
                nrows = -1 
)

AD_sumstats_4 <- read.table(file = "4_UK_Biobank_IGAP_17May2018.txt", 
                header = TRUE, 
                sep = " ", 
                skip = 0, 
                as.is = TRUE, 
                na.strings = NA, 
                comment.char = "#", 
                quote = "", 
                nrows = -1 
)

AD_sumstats <- read.table(file = "GWAS data.txt", 
                header = TRUE, 
                sep = "", 
                skip = 0, 
                as.is = TRUE, 
                na.strings = NA, 
                comment.char = "#", 
                quote = "", 
                nrows = -1 
)

library(ggplot2)
library(dplyr)



NECTIN2 <- AD_sumstats[which(AD_sumstats$Mapped.gene == "NECTIN2"), ]
NECTIN2

AC011481.1 <- AD_sumstats[which(AD_sumstats$Mapped.gene == "AC011481.1"), ]
AC011481.1

#select the samples that has p value less than 10^-6
select_sumstats <- filter(AD_sumstats_1, AD_sumstats_1$P < 0.000001)
select_sumstats <- rbind(select_sumstats, filter(AD_sumstats_2, AD_sumstats_2$P < 0.000001))

#visualization of how many SNPs has p value less than 10^-6
ggplot(select_sumstats)+
  geom_point(aes(x = SNP, y = P), color = "blue")+
  scale_y_reverse()

AD_sumstats$Allele <- gsub("?", "", gsub("\\.", "", AD_sumstats$Allele))
AD_sumstats$P.value <- as.numeric(gsub("x10", "e", gsub("\\.", "", AD_sumstats$P.value)))
AD_sumstats$Location <- as.numeric(gsub(":", ".", gsub("\\.", "", AD_sumstats$Location)))


ggplot(AD_sumstats)+
  geom_point(aes(x = Location, y = P.value), color = "red")+
  scale_y_reverse()

ggplot(AD_sumstats)+
  geom_point(aes(x = Location, y = Mapped.gene), color = "red")

ggplot(AD_sumstats, aes(x = Allele))+
  geom_bar()

```

### Results
Describe your results and include relevant tables, plots, and code/comments used to obtain them. End with a brief conclusion of your findings related to the question you set out to address. You can include references if you'd like, but this is not required.


The following 5 genes have been most frequently apppearing with a small p value (at least < 10^-5). Also a bar plot contains all the genes that has shown p value less than 10^-5 was drawn, among these genes the frequent apearing ones could be potential causative genes for AD, which worth further study.

```{r}
rev(sort(table(AD_sumstats$Mapped.gene)))[1:5]
ggplot(AD_sumstats, aes(x = Mapped.gene))+
  geom_bar()
```

Looking at the two most frequent mapped genes in the studies, 

```{r}
NECTIN2 <- AD_sumstats[which(AD_sumstats$Mapped.gene == "NECTIN2"), ]
NECTIN2

AC011481.1 <- AD_sumstats[which(AD_sumstats$Mapped.gene == "AC011481.1"), ]
AC011481.1
```

### Conclusion

Using meta-analysis with published consortium data and clinically diagnosed AD and AD-by-proxy data, 32 risk loci were identified in total, implicating 215 potential causative genes. By bringing together the results from different studies, the database was significantly enlarged. This helps us to further identify the genetic factors that contribute to AD risk based on a larger scale. The potential loci could be drug targets in the future exploration.
